tests/scripts/github-action-helper.sh

```diff
@@ -0,0 +1,184 @@
+    kubectl create -f cluster-test.yaml
+    kubectl create -f object-test.yaml
+    kubectl create -f pool-test.yaml
+    kubectl create -f filesystem-test.yaml
+    kubectl create -f rbdmirror.yaml
+    kubectl create -f nfs-test.yaml
+    kubectl create -f toolbox.yaml
+}
+
+function wait_for_prepare_pod() {
+    timeout 180 sh -c 'until kubectl -n rook-ceph logs -f $(kubectl -n rook-ceph get pod -l app=rook-ceph-osd-prepare -o jsonpath='{.items[*].metadata.name}'); do sleep 5; done' || true
+    timeout 60 sh -c 'until kubectl -n rook-ceph logs $(kubectl -n rook-ceph get pod -l app=rook-ceph-osd,ceph_daemon_id=0 -o jsonpath='{.items[*].metadata.name}') --all-containers; do echo "waiting for osd container" && sleep 1; done' || true
+    kubectl -n rook-ceph describe job/$(kubectl -n rook-ceph get pod -l app=rook-ceph-osd-prepare -o jsonpath='{.items[*].metadata.name}') || true
+    kubectl -n rook-ceph describe deploy/rook-ceph-osd-0 || true
+}
+
+function wait_for_ceph_to_be_ready() {
+    mkdir test
+    tests/scripts/validate_cluster.sh $DAEMONS $OSD_COUNT
+    kubectl -n rook-ceph get pods
+}
+
+function check_ownerreferences() {
+    curl -L https://github.com/kubernetes-sigs/kubectl-check-ownerreferences/releases/download/v0.2.0/kubectl-check-ownerreferences-linux-amd64.tar.gz -o kubectl-check-ownerreferences-linux-amd64.tar.gz
+    tar xzvf kubectl-check-ownerreferences-linux-amd64.tar.gz
+    chmod +x kubectl-check-ownerreferences
+    ./kubectl-check-ownerreferences -n rook-ceph
+}
+
+function create_LV_on_disk() {
+    BLOCK=$(sudo lsblk --paths | awk '/14G/ {print $1}' | head -1)
+    sudo sgdisk --zap-all "${BLOCK}"
+    VG=test-rook-vg
+    LV=test-rook-lv
+    sudo pvcreate "$BLOCK"
+    sudo vgcreate "$VG" "$BLOCK"
+    sudo lvcreate -l 100%FREE -n "${LV}" "${VG}"
+    tests/scripts/localPathPV.sh /dev/"${VG}"/${LV}
+    kubectl create -f cluster/examples/kubernetes/ceph/crds.yaml
+    kubectl create -f cluster/examples/kubernetes/ceph/common.yaml
+}
+
+case "$1" in
+    install_deps)
+        install_deps
+    ;;
+    print_k8s_cluster_status)
+        print_k8s_cluster_status
+    ;;
+    use_local_disk)
+        use_local_disk
+    ;;
+    use_local_disk_for_integration_test)
+        use_local_disk_for_integration_test
+    ;;
+    create_partitions_for_osds)
+        create_partitions_for_osds
+    ;;
+    create_bluestore_partitions_and_pvcs)
+        create_bluestore_partitions_and_pvcs
+    ;;
+    build_rook)
+        build_rook
+    ;;
+    validate_yaml)
+        validate_yaml
+    ;;
+    create_cluster_prerequisites)
+        create_cluster_prerequisites
+    ;;
+    deploy_cluster)
+        deploy_cluster
+    ;;
+    check_ownerreferences)
+        check_ownerreferences
+    ;;
+    create_LV_on_disk)
+        create_LV_on_disk
+    ;;
+    create_bluestore_partitions_and_pvcs_for_wal)
+        create_bluestore_partitions_and_pvcs_for_wal
+    ;;
+    wait_for_prepare_pod)
+        wait_for_prepare_pod
```

Comment:
This `case` till here directly uses the argument as the function name. We can simply the whole code (and avoid `case` condition whole together) as follows,

```bash
# assign the first argument to a common variable
selected_function="$1"
# and call the function
$selected_function
# check for any error
[ $? -ne 0 ] && echo "Function call to '$selected_function' was not successful" >&2 && exit 1
```

If you want to check some special functions, like the `wait_for_ceph_to_be_ready` function below,

```bash
# assign the first argument to a common variable
selected_function="$1"
if [ "$selected_function" = "wait_for_ceph_to_be_ready" ]; then
        # take appropriate actions/checks here
        # and then call the function with additional arguments
        $selected_function $2 $3
else
        # or else call the normal function
        $selected_function
fi
# error check will be common
[ $? -ne 0 ] && echo "Function call to '$selected_function' was not successful" >&2 && exit 1
```

Reply:
done.